exasol/analytics/query_handler/query/select.py

```diff
@@ -24,3 +28,46 @@ def __init__(self, query_string: str, output_columns: List[Column]):
     @property
     def output_columns(self) -> List[Column]:
         return self._output_columns
+
+
+class AuditData:
+    """
+    This is a collection of data for auditing. The data represent one audit message.
+    The items in the dictionary correspond to the columns in the audit table.
+    Components at different levels in the call stack can add their own items here.
+    """
+    def __init__(self, audit_fields: dict[str, Any] | None = None):
+        self._audit_fields = audit_fields or {}
+
+    @property
+    def audit_fields(self) -> dict[str, Any]:
+        return self._audit_fields
+
+
+class AuditQuery(AuditData):
```

Comment:
```suggestion
class AuditQuery(AuditData, Query):
```

this way, we can inject the AuditQuery without problems into the query_list of Continue. FYI, Query is an interface that requests to implement the property query_text, which would be in this case simply the forwarding of the query_text of the _query_with_columns

Reply:
this also guarantees backwards compatibility with the Lua QueryHandler Runner which would run this query as is without doing logging. Because, audit logging, we only will add after Q2 to the Lua QueryHandler Runner.

Reply:
```
def query_text():
    if self._select_query_with_columns is not None:
        return self._select_query_with_columns.query_text
   else:
       return "SELECT 1"
```